authors/apps/authentication/tests/test_email_verification.py

```diff
@@ -0,0 +1,44 @@
+from django.urls import reverse
+from rest_framework.test import APIClient
+from rest_framework.test import APITestCase
+from rest_framework import status
+
+
+class TestEmailVerify(APITestCase):
+    '''class for testing email verification'''
+
+    def setUp(self):
+        """
+        Prepare test environment for each testcase
+        """
+        self.client = APIClient()
+        self.signup_url = reverse('authentication:registration')
+        self.user_details = {
+            'user': {
+                'username': 'user1',
+                'email': 'evajohnson714@gmail.com',
+                'password': 'somepass12345',
+            }
+        }
+
+    def test_account_verification(self):
+        """
+        Test whether account was verified in success.
+        """
+        self.client.post(self.signup_url, self.user_details, format='json')
+        response = self.client.get(
+            reverse(
+                "authentication:verify",
+                args=['sometoken']))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_account_verification_with_wrong_token(self):
+        """
+        Test whether account can be verified with inavlid token.
+        """
+        self.client.post(self.signup_url, self.user_details, format='json')
+        response = self.client.get(
+            reverse(
+                "authentication:verify",
+                args=['aidhugjhxdvhjv']))
```

Comment:
Do you check whether the email and username are correct?

Reply:
> Do you check whether the email and username are correct?

We are not checking whether the email or username is correct. That should be done on the registration feature. Ours is just to send the verification link to activate account when a user registers